### DISCUSSION TITLE
Is there any way to prevent Line break?

### DISCUSSION THREAD

**Discussion by rulai-yinqiang** [harborsiem/WindowsRibbon#27] [Q&A] [answered]
If I set 2 Chinese characters in Large button, each Chinese characters is line breaked. But Windows Explorer looks good. Can I set the button width or prevent line break?

![image](https://github.com/harborsiem/WindowsRibbon/assets/33405365/c77a64ef-0b6b-4557-9e21-051e9a58d9bf)

**Comment by harborsiem** [comment]
Is the Line break done for the Label of the large button? 
What happend when you use this character in Tooltips for this button ? Is there also a line break?
Did you know the binary representation of this character ?

**Comment by rulai-yinqiang** [comment]
Any 2 Chinese characters will break. For example, I change the WordPad exmaple Paste label to '粘贴', save, build, preview. The screen attached.
BTW: I'm testing in Win11 Chinese version.
![image](https://github.com/harborsiem/WindowsRibbon/assets/33405365/85b8fd55-9876-439b-8590-c64bdd39ca41)

**Comment by harborsiem** [comment]
May be there is a solution for this problem (I think it is a bug in the Microsoft implementation).

Open the RibbonMarkup.xml with a text editor. Add character sequence **&#xA** **;** without the space between A and ; (this is a line break in xml) at the end of the Chinese characters.
Open RibbonMarkup.xml with RibbonTools again. Build and Preview.
If this is helpful, then I will do a change in the RibbonTools.

**Comment by rulai-yinqiang** [comment]
Thanks for the tips, but it seems it's not work.
![image](https://github.com/harborsiem/WindowsRibbon/assets/33405365/e5bbbea6-4003-4921-8a2a-f288bc401489)

**Comment by harborsiem** [comment, marked as answer]
Open the RibbonMarkup.xml with a text editor.
text editor should not be the RibbonTools. You can use for example Notepad++. 

Here is an example with emoji's

![Notepad_Edit](https://github.com/harborsiem/WindowsRibbon/assets/28147549/8290268d-36da-4af6-a0c2-f9aa80b3b57e)

![RibbonTools](https://github.com/harborsiem/WindowsRibbon/assets/28147549/b417731e-0945-4326-8f9d-b27c285ff21a)

![Preview](https://github.com/harborsiem/WindowsRibbon/assets/28147549/cb2f5862-84bc-4cbf-b0f1-dcb22ba39160)

**Comment by rulai-yinqiang** [comment]
Yes! Add a Line Break at the end of Label works! Thanks.

**Comment by harborsiem** [comment]
I have done a Pre-Release v2.15.2 to solve the problem. Now you can also edit in RibbonTools the Line break at the end of Label with **\n**
